src/app/shared/role-guard-service.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@angular/core';
+import { CouchService } from './couchdb.service';
+import { UserService } from './user.service';
+import { Router, CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot } from '@angular/router';
+
+@Injectable()
+export class RoleService {
+
+  constructor(private couchService: CouchService, private userService: UserService, private router: Router) {}
+
+  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): any {
+    if (route.url.toString() === 'add') {
```

Comment:
We do not need to check the route here, because this only applies to the routes we place it on in the router files.